src/index.js

```diff
@@ -252,20 +252,21 @@ fflags.enabled('onetrust', () => {
     pinterest: /epik/,
     tiktok: /ttclid/,
   };
-  const params = Array.from(new URLSearchParams(window.location.search).keys());
-  Object.entries(networks).forEach(([network, regex]) => {
-    params.filter((param) => regex.test(param)).forEach((param) => sampleRUM('paid', { source: network, target: param }));
-  });
-})();
 
-fflags.enabled('email', () => {
-  const networks = {
+  const emailNetworks = {
     mailchimp: /mc_(c|e)id/,
     marketo: /mkt_tok/,
 
   };
-  const params = Array.from(new URLSearchParams(window.location.search).keys());
-  Object.entries(networks).forEach(([network, regex]) => {
-    params.filter((param) => regex.test(param)).forEach((param) => sampleRUM('email', { source: network, target: param }));
-  });
-});
+
+  const knownAdNetworksSection = matchParams(adNetworks, searchParams).map((network) => `paid=${network}`);
+  const knownEmailNetworksSection = matchParams(emailNetworks, searchParams).map((network) => `email=${network}`);
+  const utmSection = searchParams.filter(([key]) => key.startsWith('utm_'))
+    .filter(([key]) => key !== 'utm_id')
+    .filter(([key]) => key !== 'utm_term')
+    .map(([key, value]) => `${key}=${value}`);
```

Comment:
```suggestion
    .map(([_, value]) => .value).join();
```

I don't distinguish between source and medium on the collector side, so we can leave that out.